src/wallet/rpcdump.cpp

```diff
@@ -157,8 +157,8 @@ UniValue importprivkey(const JSONRPCRequest& request)
         if (!request.params[2].isNull())
             fRescan = request.params[2].get_bool();
 
-        if (fRescan && pwallet->chain().getPruneMode()) {
-            throw JSONRPCError(RPC_WALLET_ERROR, "Rescan is disabled in pruned mode");
+        if (fRescan && pwallet->chain().havePruned()) {
```

Comment:
A block can be pruned before RescanWallet (L202). What would be the error then?

Reply:
It will throw an error `"Rescan was unable to fully rescan the blockchain. Some transactions may be missing."`